ss58-registry-derive/src/lib.rs

```diff
@@ -88,249 +71,292 @@ impl Registry {
-        if name.ends_with("net") {
-            name.truncate(name.len() - 3);
-        }
-        // Capitalise
-        name.get_mut(0..1)
-            .expect("name should not be empty")
-            .make_ascii_uppercase();
-
-        let postfix = if is_reserved { "" } else { "Account" };
-        format!("{}{}", rust_valid_id(name), postfix)
+        format!(
+            "{}Account",
+            inflector::cases::pascalcase::to_pascal_case(&self.network)
+        )
     }
 }
 
-fn rust_valid_id(name: String) -> String {
-    // TODO find function that already does this. `-` are excluded in particular.
-    name.chars().filter(|c| c.is_alphanumeric()).collect()
-}
-
 /// Creates the Ss58AddressFormat enum from the ss58-registry.json file
 #[proc_macro]
 pub fn ss58_registry_derive(input: TokenStream) -> TokenStream {
     assert!(input.is_empty(), "No arguments are expected");
 
-    match create_ss58_registry(include_str!("ss58-registry.json")) {
-        Ok(result) => result,
+    match create_ss58_registry(include_str!("../../ss58-registry.json")) {
+        Ok(result) => result.into(),
         Err(msg) => panic!("{}", msg),
     }
 }
 
-fn create_ss58_registry(json: &str) -> Result<TokenStream, String> {
+fn create_ss58_registry(json: &str) -> Result<proc_macro2::TokenStream, String> {
     let registry: Registry = serde_json::from_str(json).expect("valid json file");
-
     registry.is_valid()?;
 
+    let registry = registry.registry;
+
+    // Variables to insert into quote template:
     let identifier: Vec<_> = registry
-        .registry
         .iter()
         .map(|r| format_ident!("{}", r.name()))
         .collect();
 
-    let reserved_identifiers: Vec<_> = registry
-        .registry
+    let reserved_identifiers = registry
         .iter()
-        .filter(|r| {
-            r.network
-                .as_ref()
-                .filter(|r| r.starts_with("reserved"))
-                .is_some()
-        })
-        .map(|r| format_ident!("{}", r.name()))
-        .collect();
+        .filter(|r| r.standard_account.is_none())
+        .map(|r| format_ident!("{}", r.name()));
 
-    let reserved_numbers: Vec<_> = registry
-        .registry
+    let reserved_numbers = registry
         .iter()
-        .filter(|r| {
-            r.network
-                .as_ref()
-                .filter(|r| r.starts_with("reserved"))
-                .is_some()
-        })
-        .map(|r| r.prefix)
-        .collect();
+        .filter(|r| r.standard_account.is_none())
+        .map(|r| r.prefix);
 
-    let number: Vec<_> = registry.registry.iter().map(|r| r.prefix).collect();
-    let count = registry.registry.len();
-    let name: Vec<_> = registry
-        .registry
-        .iter()
-        .map(|r| r.network.clone().unwrap_or_else(|| "Bare".to_string()))
-        .collect();
-    let desc = registry.registry.iter().map(|r| {
+    let count = registry.len();
+    let number: Vec<_> = registry.iter().map(|r| r.prefix).collect();
```

Comment:
```suggestion
    let number = registry.iter().map(|r| r.prefix);
```

Reply:
Again it's iterated over more than once.